pyfar/classes/coordinates.py

```diff
@@ -2807,12 +2807,146 @@ def __eq__(self, other):
+
+    def __rmul__(self, other):
+        """Multiply two Coordinates objects."""
+        return _arithmetics(other, self, 'mul')
+
+    def __div__(self, other):
+        """Divide one Coordinates objects."""
+        return _arithmetics(self, other, 'div')
+
+    def __truediv__(self, other):
+        """Divide one Coordinates objects."""
+        return _arithmetics(self, other, 'div')
+
+    def __rtruediv__(self, other):
+        """Divide two Coordinates objects."""
+        return _arithmetics(other, self, 'div')
+
+    def __rdiv__(self, other):
+        """Divide two Coordinates objects."""
+        return _arithmetics(other, self, 'div')
+
+    def dot(self, other):
+        """Dot product two Coordinates objects."""
+
+        if not isinstance(other, Coordinates):
+            raise TypeError(
+                "Dot product is only possible with Coordinates objects.")
+
+        return self.x * other.x + self.y * other.y + self.z * other.z
+
+
+    def cross(self, other):
+        """Cross product two Coordinates objects
+
+        Parameters
+        ----------
+        other : pf.Coordinates
+            other Coordinates object to perform the cross product with
+
+        Returns
+        -------
+        result : pf.Coordinates
+            new Coordinates object with the cross product of the two objects
+        """
+
+        if not isinstance(other, Coordinates):
+            raise TypeError(
+                "Dot product is only possible with Coordinates objects.")
+
+        new = Coordinates()
+        new.cartesian = np.zeros(np.broadcast_shapes(
+            self.cartesian.shape, other.cartesian.shape))
+
+        # apply cross product
+        new.x = self.y * other.z - self.z * other.y
+        new.y = self.z * other.x - self.x * other.z
+        new.z = self.x * other.y - self.y * other.x
+
+        return new
+
+
     def _check_empty(self):
         """check if object is empty"""
         if self.cshape == (0,):
             raise ValueError('Object is empty.')
 
 
+def _arithmetics(first, second, operation):
+    """Add or Subtract two Coordinates objects, numbers or arrays.
+
+    Parameters
+    ----------
+    first : Coordinates, number, array
+        first operand
+    second : Coordinates, number, array
+        second operand
+    operation : 'add', 'sub', 'mul', 'div'
+        whether to add or subtract the two objects
+
+    Returns
+    -------
+    new : Coordinates
+        result of the operation
+
+    """
+    # convert data
+    data = []
+    num_objects = 0
+    for obj in [first, second]:
+        if isinstance(obj, Coordinates):
+            data.append(obj.cartesian)
+            num_objects += 1
+        elif isinstance(obj, (int, float)):
+            data.append(np.array(obj))
+        else:
+            op = 'Addition' if operation == 'add' else 'Subtraction'
+            raise TypeError(
+                f"{op} is only possible with Coordinates or number.")
```

Comment:
This could also be reached for operations 'mul' and 'div' - why are they not considered for throwing the error? In addition, the docstring states that arrays can be used, but currently I think this would raise the TypeError as well. Maybe we can discuss if there are use cases for array input, or if any array data must be passed as coordintates.